airflow/models.py

```diff
@@ -2463,6 +2463,29 @@ def get_many(
         return query.all()
 
 
+class DagRun(Base):
+    """
+    DagRun describes an instance of a Dag. It can be created
+    by the scheduler (for regular runs) or by an external trigger
+    """
+    __tablename__ = "dag_run"
+
+    dag_id = Column(String(ID_LEN), primary_key=True)
+    execution_date = Column(DateTime, primary_key=True)
+    run_id = Column(String(ID_LEN))
+    external_trigger = Column(Boolean, default=False)
```

Comment:
we may want to add a state here, so that the scheduler could completely disregard DAGs that are fully processed. I'm not sure whether it should just be boolean or if having more states would help.